graph/Force-Directed-Graph/ForceDirectedGraph.tsx

```diff
@@ -0,0 +1,171 @@
+  constrainedNodeId: string;
+}> = ({ graph, constrainedNodeId }) => {
+  // States to store the nodes, links, sizes and loading status
+  const [nodes, setNodes] = useState<Node[]>([]);
+  const [links, setLinks] = useState<Link[]>([]);
+  const [sizes, setSizes] = useState<Map<string, number>>(new Map());
+  const [load, setLoad] = useState<boolean>(false);
+
+  // Use effect to update the graph
+  useEffect(() => {
+    // Get the initial links, nodes and sizes
+    const initialLinks = graph.getLinks();
+    const initialNodes = graph.getNodes();
+    const initialSizes = setNodesSizes([...initialLinks]);
+    // Set the links, nodes and sizes
+    setLinks([...initialLinks]);
+    setNodes(
+      fruchtermanReingold(
+        [...initialNodes],
+        initialLinks,
+        constrainedNodeId,
+        WIDTH,
+        HEIGHT,
+        MAX_ITERATIONS
+      )
+    );
+    setSizes(initialSizes);
+    setLoad(true);
+  }, [graph, constrainedNodeId]);
+
+  // If the graph is not loaded, display an activity indicator
+  if (!load) {
+    return <ActivityIndicator size="large" color="#0000ff" />;
+  }
+
+  // Functions to get the X and Y coordinates of the nodes and the fill color of the nodes
+  const coordX = (node: Node): number => {
+    return node.x;
+  };
+
+  const coordY = (node: Node): number => {
+    return node.y;
+  };
+
+  const circleFill = (node: Node): string => {
+    if (node.id === constrainedNodeId) {
+      return CONSTRAINED_NODE_COLOR;
+    }
+    return DEFAULT_NODE_COLOR;
+  };
+
+  // Create the lines and circles for the graph
+  const LINES = links.map((link) => (
+    <Line
+      key={link.source + link.target}
+      x1={coordX(
+        nodes.find((node) => node.id === link.source) ?? {
+          x: CENTER_WIDTH,
+          y: CENTER_HEIGHT,
+          id: link.source,
+          dx: 0,
+          dy: 0,
+        }
+      )}
+      y1={coordY(
+        nodes.find((node) => node.id === link.source) ?? {
+          x: CENTER_WIDTH,
+          y: CENTER_HEIGHT,
+          id: link.source,
+          dx: 0,
+          dy: 0,
+        }
+      )}
+      x2={coordX(
+        nodes.find((node) => node.id === link.target) ?? {
+          x: CENTER_WIDTH,
+          y: CENTER_HEIGHT,
+          id: link.target,
+          dx: 0,
+          dy: 0,
+        }
+      )}
+      y2={coordY(
+        nodes.find((node) => node.id === link.target) ?? {
+          x: CENTER_WIDTH,
+          y: CENTER_HEIGHT,
+          id: link.target,
+          dx: 0,
+          dy: 0,
+        }
+      )}
+      stroke={DEFAULT_LINK_COLOR}
+    />
+  ));
+
+  const CIRCLES = nodes.map((node) => (
+    <Circle
+      key={node.id}
+      cx={coordX(node)}
+      cy={coordY(node)}
+      r={sizes.get(node.id) ?? DEFAULT_NODE_SIZE}
+      fill={circleFill(node)}
+    />
+  ));
+
+  return (
+    <View style={styles.container}>
+      <Svg width={WIDTH} height={HEIGHT}>
+        {LINES}
+        {CIRCLES}
+      </Svg>
+    </View>
+  );
```

Comment:
Are you forced to use svg ? as we stated that we could use easier alternatives

Reply:
We talked about using **React-Native-Skia.** From what I have seen people seem to really like it. However, my understanding it that it is very useful for advanced things after testing it a bit **react-native-svg** is **_easier_** to use for the current implementation of the graph. Thus, I don't think it is relevant (at least for now) to use Skia over Svg. In the event where some later implementations require something very specific and only achievable by Skia then I will update the code but note that this is very unlikely, especially for our V1.

You can find the documentation for both libraries below:
- [React-Native-Skia](https://shopify.github.io/react-native-skia/docs/getting-started/installation)
- [React-Native-Svg](https://github.com/software-mansion/react-native-svg)